cumulusci/core/tests/test_tasks.py

```diff
@@ -59,6 +59,35 @@ def setUp(self):
         self._task_log_handler.reset()
         self.task_log = self._task_log_handler.messages
 
+    @mock.patch("cumulusci.core.tasks.time.sleep", mock.Mock())
+    def test_retry_on_exception(self):
+        """ calling _retry() should call try until the task succeeds.  """
+        task = BaseTask(self.project_config, self.task_config, self.org_config)
+        task.options = {"retries": 5, "retry_interval": 1, "retry_interval_add": 1}
+        task._try = mock.Mock(side_effect=[Exception, Exception, 1])
+        task._retry()
+        self.assertEqual(task._try.call_count, 3)
+
+    @mock.patch("cumulusci.core.tasks.time.sleep", mock.Mock())
+    def test_retry_until_too_many(self):
+        """ calling _retry should call try until the retry count is exhausted. """
+        task = BaseTask(self.project_config, self.task_config, self.org_config)
+        task.options = {"retries": 5, "retry_interval": 1, "retry_interval_add": 1}
+        task._try = mock.Mock(
+            side_effect=[
+                RuntimeError(5),
+                RuntimeError(4),
+                RuntimeError(3),
+                RuntimeError(2),
+                RuntimeError(1),
+                Warning,
+            ]
+        )
+        with self.assertRaises(Warning):
+            task._retry()
```

Comment:
I think it would be clearer what this was testing if you used the same class of exception for the last one (and assert that the one with the correct `args` was raised). `Warning` in particular made me go do a minute of research because I hadn't seen it before and wanted to double check that it is in fact a subclass of Exception!

Reply:
cool! i was hoping this was more obvious, but upon reflection I think your suggestion of just checking the args is good.